### PR TITLE
Gemv implementation

### PR DESCRIPTION
**Pull request by avolkov-intel**
Add interface for mkl::blas::gemv

### CODE REVIEW

File: cpp/oneapi/dal/backend/primitives/blas/gemv_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                                                   const event_vector& deps);
+
+#define INSTANTIATE_FLOAT(ao) \
+    INSTANTIATE(float, ao)    \
+    INSTANTIATE(double, ao)
+
+INSTANTIATE_FLOAT(ndorder::c);
+INSTANTIATE_FLOAT(ndorder::f);
+
+} // namespace oneapi::dal::backend::primitives
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/blas/gemv_dpc.cpp:91]: Please add new line

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    SKIP_IF(this->get_policy().is_cpu());
+
+    // Test takes too long time if HW emulates float64
+    SKIP_IF(this->not_float64_friendly());
+
+    this->generate_medium_dimensions();
+    this->test_gemv();
+}
+
+} // namespace oneapi::dal::backend::primitives::test
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:174]: EOL

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        SUCCEED();
+    }
+
+    bool is_initialized() const {
+        return m_ > 0 && n_ > 0;
+    }
+
+    void check_if_initialized() {
+        if (!is_initialized()) {
+            throw std::runtime_error{ "gemm test is not initialized" };
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:141]:
```suggestion
            throw std::runtime_error{ "gemv test is not initialized" };
```

**Review by KulikovNikita — COMMENTED** [review]
Please fix minor comments

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+private:
+    std::int64_t m_;
+    std::int64_t n_;
+};
+
+using gemv_types = COMBINE_TYPES((float, double), (c_order, f_order));
+
+TEMPLATE_LIST_TEST_M(gemv_test, "ones matrix gemv on small sizes", "[gemv][small]", gemv_types) {
+    // DPC++ GEMM from micro MKL libs is not supported on GPU
```
Comment (Alexsandruss) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:153]: DPC++ GEMM not GEMV?

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    // Test takes too long time if HW emulates float64
+    SKIP_IF(this->not_float64_friendly());
+
+    this->generate_small_dimensions();
+    this->test_gemv();
+}
+
+TEMPLATE_LIST_TEST_M(gemv_test, "ones matrix gemv on medium sizes", "[gemv][small]", gemv_types) {
+    // DPC++ GEMM from micro MKL libs is not supported on GPU
+    SKIP_IF(this->get_policy().is_cpu());
```
Comment (Alexsandruss) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:165]: It is not supported on GPU but `SKIP_IF(... .is_cpu())`?

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+            gemv(this->get_queue(), at.t(), x, y_ones, alpha, beta, { a_e, x_e, y_ones_e })
+                .wait_and_throw();
+            check_value_vector(y_ones, m_, n_ * alpha + beta);
+        }
+    }
+
+    void check_value_vector(const ndarray<float_t, 1>& vec, int64_t len, float_t value) {
+        check_if_initialized();
+        // const float_t* ptr2 = vec.get_data();
```
Comment (Alexsandruss) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:122]: Remove comment or add explanation above

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <type_traits>
+
+#include "oneapi/dal/backend/primitives/blas/gemv.hpp"
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include <iostream>
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:22]: You don't need this header

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp (hunk 4 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:153]:
```suggestion
    // DPC++ GEMV from micro MKL libs is not supported on GPU
```

File: cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp (hunk 5 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp:165]: > It is not supported on GPU but `SKIP_IF(... .is_cpu())`?

Nope. It is not. AFAIK on CPU if should fall back to the conventional MKL GEMV implementation

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**